website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -5431,4 +5431,282 @@ use case.
+
+Python has a default expansion service included and available in the Apache Beam SDK. You are free to write your own expansion service, but that is generally not needed, so it is not covered in this section.
+
+Perform the following steps to start up the default Python expansion service directly:
+
+1. Create a virtual environment and [install the Apache Beam SDK](https://beam.apache.org/get-started/quickstart-py/).
+2. Start the Python SDK’s expansion service with a specified port.
+
+    {{< highlight >}}
+$ export PORT_FOR_EXPANSION_SERVICE=12345
+$ python -m apache_beam.runners.portability.expansion_service -p $PORT_FOR_EXPANSION_SERVICE
+    {{< /highlight >}}
+
+**Including dependencies**
+
+If your transform requires external libraries, make sure those dependencies are already specified in a custom container for SDK harness. Core Apache Beam dependencies are already included.
+
+### 13.2. Using cross-language transforms {#use-x-lang-transforms}
+
+Depending on the SDK language of the pipeline, you can use a high-level SDK-wrapper class, or a low-level transform class to access a cross-language transform. 
+
+#### 13.2.1. Using cross-language transforms in a Java pipeline
+
+Currently, to access cross-language transforms from the Java SDK, you have to use the lower-level [External](https://github.com/apache/beam/blob/master/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/External.java) class.
+
+**Using the External class**
+
+1. Make sure you have any runtime environment dependencies (like JRE) installed on your local machine (either directly on the local machine or available through a container). See the expansion service section for more details.
+
+    > **Note:** When including Python transforms from within a Java pipeline, all python dependencies have to be baked into the SDK harness container.
+2. Start up the expansion service for the SDK that is in the language of the transform you're trying to consume, if not available. 
+
+    Make sure the transform you are trying to use is available and can be used by the expansion service.
+3. Include [External.of(...)](https://github.com/apache/beam/blob/master/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/External.java) when instantiating your pipeline. Reference the URN, payload, and expansion service. For examples, see the [cross-language transform test suite](https://github.com/apache/beam/blob/master/runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ValidateRunnerXlangTest.java).
+4. After the job has been submitted to the Beam runner, shutdown the expansion service by terminating the expansion service process.
+
+#### 13.2.2 Using cross-language transforms in a Python pipeline
+
+If a Python-specific wrapper for a cross-language transform is available, use that; otherwise, you have to use the lower-level `ExternalTransform` class to access the transform.
```

Comment:
Add a link to https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/external.py

Reply:
Done.